cmd/daemon/kubernetes/kubernetes.go

```diff
@@ -27,7 +29,15 @@ var (
 )
 
 func NewClient() (*Client, error) {
-	config, err := rest.InClusterConfig()
+	var config *rest.Config
+	var err error
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		config, err = rest.InClusterConfig()
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	}
+
```

Comment:
This could be simplified a bit as `BuildConfigFromFlags` [falls back to using `InClusterConfig`](https://godoc.org/k8s.io/client-go/tools/clientcmd#BuildConfigFromFlags).
```suggestion
	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
```

Reply:
Nice, didn't know that 💪